dds/DCPS/DataReaderImpl.cpp

```diff
@@ -581,13 +581,8 @@ DataReaderImpl::remove_or_reschedule(const PublicationId& pub_id)
     WriterIdSeq writers;
     push_back(writers, pub_id);
     bool notify = info.notify_lost_;
-    if (info.removal_deadline_ < ACE_OS::gettimeofday()) {
-      write_guard.release();
-      remove_associations_i(writers, notify);
-    } else {
-      write_guard.release();
-      remove_associations(writers, notify);
-    }
+    write_guard.release();
```

Comment:
Even though the explicit release() was there before, if you have time please update this function to remove it.  Seems like this one would be easy enough to move outside of the scope of the lock and move the locals above it.  Also a great illustration of why we prefer multiple exit points from a function.